java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -499,461 +225,212 @@ public boolean equals(Object obj) {
     return this == obj;
   }
 
-  @Override
-  public ArrowBuf retain(int increment) {
-    Preconditions.checkArgument(increment > 0, "retain(%d) argument is not positive", increment);
-
-    if (isEmpty) {
-      return this;
-    }
-
-    if (BaseAllocator.DEBUG) {
-      historicalLog.recordEvent("retain(%d)", increment);
-    }
-
-    final int originalReferenceCount = refCnt.getAndAdd(increment);
-    Preconditions.checkArgument(originalReferenceCount > 0);
-    return this;
-  }
-
-  @Override
-  public ArrowBuf retain() {
-    return retain(1);
-  }
-
-  @Override
-  public ByteBuf touch() {
-    return this;
+  /**
+   * IMPORTANT NOTE
+   * The data getters and setters work with a caller provided
+   * index. This index is 0 based and since ArrowBuf has access
+   * to a portion of underlying chunk of memory starting at
+   * some address, we convert the given relative index into
+   * absolute index as memory address + index.
+   *
+   * Example:
+   *
+   * Let's say we have an underlying chunk of memory of length 64 bytes
+   * Now let's say we have an ArrowBuf that has access to the chunk
+   * from offset 4 for length of 16 bytes.
+   *
+   * If the starting virtual address of chunk is MAR, then memory
+   * address of this ArrowBuf is MAR + offset -- this is what is stored
+   * in variable addr. See the BufferLedger and AllocationManager code
+   * for the implementation of ReferenceManager that manages a
+   * chunk of memory and creates ArrowBuf with access to a range of
+   * bytes within the chunk (or the entire chunk)
+   *
+   * So now to get/set data, we will do => addr + index
+   * This logic is put in method addr(index) and is frequently
+   * used in get/set data methods to compute the absolute
+   * byte address for get/set operation in the underlying chunk
+   *
+   * @param index the index at which we the user wants to read/write
+   * @return the absolute address within the memro
+   */
+  private long addr(int index) {
+    return addr + index;
   }
 
-  @Override
-  public ByteBuf touch(Object hint) {
-    return this;
+  public long getLong(int index) {
+    chk(index, 8);
+    return PlatformDependent.getLong(addr(index));
   }
 
-  @Override
-  public long getLong(int index) {
+  public ArrowBuf setLong(int index, long value) {
     chk(index, 8);
-    final long v = PlatformDependent.getLong(addr(index));
-    return v;
+    PlatformDependent.putLong(addr(index), value);
+    return this;
   }
 
-  @Override
   public float getFloat(int index) {
```

Comment:
generally same question about changing the API to use Longs instead of Ints for all getters/settings